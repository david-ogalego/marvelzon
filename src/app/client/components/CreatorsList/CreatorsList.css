.creatorsList {
    margin-top: 145px;
    padding-right: 5px;
    padding-left: 5px;
}