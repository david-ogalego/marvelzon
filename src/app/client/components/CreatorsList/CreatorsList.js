import React from 'react';
import stylesCreatorList from './CreatorsList.css';

const CreatorsList = function(props) {
    return (
        <div className={stylesCreatorList.creatorsList}>
            <h3>Creators</h3>
            <ul>
                {props.creators.map((creator, index) => 
                    <li key={index}>
                        <span>Name: {creator.name}</span>
                        <span>Role: {creator.role}</span>
                    </li>
                )
            }
            </ul>
        </div>
    );
}

export default CreatorsList;