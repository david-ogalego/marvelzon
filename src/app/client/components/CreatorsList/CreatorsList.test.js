import React from 'react';
import CreatorsList from './CreatorsList';
import renderer from 'react-test-renderer';

const creatorsList = [
    {
        name: 'David',
        role: 'writer'
    },
    {
        name: 'Pepe',
        role: 'penciller (cover)'
    }
];

test('Snapshot CreatorsList rendered empty', () => {
    const treeCreatorsListEmpty = renderer.create(<CreatorsList creators={[]} />).toJSON();
    expect(treeCreatorsListEmpty).toMatchSnapshot();
});

test('Snapshot CreatorsList rendered with creators', () => {
    const treeCreatorsList = renderer.create(<CreatorsList creators={creatorsList} />).toJSON();
    expect(treeCreatorsList).toMatchSnapshot();
});