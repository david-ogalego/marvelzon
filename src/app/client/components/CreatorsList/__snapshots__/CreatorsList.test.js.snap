// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot CreatorsList rendered empty 1`] = `
<div
  className={undefined}
>
  <h3>
    Creators
  </h3>
  <ul />
</div>
`;

exports[`Snapshot CreatorsList rendered with creators 1`] = `
<div
  className={undefined}
>
  <h3>
    Creators
  </h3>
  <ul>
    <li>
      <span>
        David
         - 
        writer
      </span>
    </li>
    <li>
      <span>
        Pepe
         - 
        penciller (cover)
      </span>
    </li>
  </ul>
</div>
`;
